concepts/modules/about.md

```diff
@@ -0,0 +1,101 @@
+# Modules
+
+[Modules][modules] in Crystal serves 2 purposes:
+
+The first purpose is to use it as a [namespace][namespace], this is mainly to avoid name clashes.
+But it also forms as a form of grouping code together, this is to make it easier to understand what the code is for.
+
+The second purpose is to use it as a [mixin][mixin], this is to share code to types.
+
+Modules has similarities to classes, but the main difference is that modules can not be instantiated, and thereby don't have instance variables.
+
+To declare a module you use the `module` keyword, followed by the name of the module.
+
+```crystal
+module Foo
+end
+```
+
+## Namespace
+
+A namespace is a way to group code together, this is to avoid name clashes, but also to make it easier to understand what the code is for.
+When wanting to access for example a constant or a class that has been placed inside a namespace you use the `::` operator.
+
+```crystal
+module Foo
+  class Bar
+  end
+end
+
+Foo::Bar.new
+```
+
+## Use it as a mixin
+
+A mixin is a way for another type to be able to use the code inside the module.
+This can be useful when for example wanting multiple classes to have the same "base" functionality.
+Or when wanting to share code between classes that are not related.
+
+There are 2 different ways to use a module as a mixin, the first one is to use the `include` keyword, the second one is to use the `extend` keyword.
+
+Both methods will make constants available to the type that includes or extends the module.
+
+### Include
+
+Include will make all methods in the module available as instance methods on the type that includes the module.
+The `include` keyword should be written at the top of the type, followed by the name of the module.
+
+```crystal
+module Foo
+  def foo
+    "foo"
+  end
+end
+
+class Bar
+  include Foo
+end
+
+Bar.new.foo # => "foo"
+```
+
+### Extend
+
+Extend works similarly to include, but instead of making the methods available as instance methods, it will make them available as class methods.
```

Comment:
```suggestion
Extend works similarly to include, but instead of making the methods available as instance methods, it makes them available as class methods.
```